draft-add-requirements.md

```diff
@@ -55,66 +92,119 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
 document are to be interpreted as described in BCP 14 {{RFC2119}} {{!RFC8174}}
 when, and only when, they appear in all capitals, as shown here.
 
-# Discovery of associated resolvers {#first-usecase}
-
+{: #first-usecase}
+# Discovery of associated resolvers
 A client may begin with information from the attached networks, and/or resolvers known from configuration.
 This information may be used to then discover one or more associated or equivalent encrypted resolvers.
 These are expected to be operated by the same entity that provides the resolver the client started with, but
-may not have identical behaviour. For example one entity may offer a set of encrypted resolvers with different levels of
-filtering or proximity.
+may not have identical behaviour. For example one entity may offer a set of encrypted resolvers with different levels
+of filtering or proximity.
+
+These resolver upgrade mechanisms can either add new parameters to existing provisioning mechanisms
+(adding necessary information to use DoT or DoH to options in DHCP, RAs, or IKEv2) or else provide a way
+to communicate with a provisioned unencrypted DNS resolver and discover the equivalent or associated encrypted DNS resolver.
 
 | Requirement | Description |
 | R1.1 | There must be a mechanism for a client to learn the set of encrypted resolvers that are associated with a resolver that is known only by its IP address. |
```

Comment:
As far as flow, the way we had tried to lay things out was to derive the mechanism requirements after describing the use cases. We can certainly refer back to the use cases that motivate requirements.

My question reading this requirement up front like this (before getting to use cases) is "why only the IP address"? If it was only about network provisioning, this might not be a requirements, and I think this one comes more strictly from the user-selected resolver case.

Reply:
Agree with Tommy, we should focus on generic requirements without hinting on possible solutions. We may want to re-phrase the requirements as follows:
 R1.1 There must be a mechanism for a client to learn the set of encrypted resolver that are associated with a unencrypted resolver.
 R1.2 | Discovery must be possible even when the IP address of the encrypted DNS resolver is only valid locally.

Reply:
Agree that IP address is too specific here. I'm happy with Tiru's rephrasing.